Enabling building llama.cpp using system libggml

ckastner opened the pull request:
A llama.cpp build could make use of an already installed libggml using the latter's cmake package. This would be the typical case for Linux distributions.

Note: This requires a ggml build where GGML_BIN_DIR is commented out, see [82d9ca4](https://github.com/ggml-org/ggml/commit/82d9ca4c0917427ab23195af768b588d21a220f7).

```shell
# Build and install ggml to some custom path
$ git clone https://github.com/ggml-org/ggml.git && cd ggml
$ cmake -B build -DCMAKE_INSTALL_PREFIX=/tmp/system-ggml -DCMAKE_INSTALL_RPATH='$ORIGIN'
$ cmake --build build && cmake --install build

# Build and install llama against this ggml, using ggml's cmake package
$ cmake -B build \
	-DLLAMA_USE_SYSTEM_GGML=ON \
	-DCMAKE_PREFIX_PATH=/tmp/system-ggml/lib/cmake/ggml \
	-DLLAMA_BUILD_EXAMPLES=ON \
	-DLLAMA_BUILD_TESTS=ON
$ cmake --build build
$ build/bin/llama-bench
...
$ bash ./ci/run.sh ./tmp/results ./tmp/mnt
...

# Negative check, to make sure we didn't break the regular build
$ cmake -B build \
	-DLLAMA_USE_SYSTEM_GGML=OFF \
	-DLLAMA_BUILD_EXAMPLES=ON \
	-DLLAMA_BUILD_TESTS=ON
$ cmake --build build
...
$ bash ./ci/run.sh ./tmp/results ./tmp/mnt
...
```

The tests passed but I get a `ModuleNotFoundError` (torch) from `convert_hf_to_gguf.py`, which is unexpected. I haven't  investigated yet.

ckastner committed:
cmake: Factor out compiler flag function from ggml

llama.cpps's build requires it, too, and we may want to make use of it
without add_subdirectory(ggml).

Green-Sky commented on CMakeLists.txt:
```diff
@@ -145,7 +147,15 @@ endif()
+    find_package(ggml REQUIRED)
+    set(GGML_TARGET ggml::ggml)
+else()
+    set(GGML_TARGET ggml)
```
instead of using a `GGML_TARGET` var, we should just add the `ggml::ggml` alias target

ckastner replied:
That indeed simplifies the approach.

ckastner committed:
cmake: Enable building against system ggml

This facilitates package maintenance for Linux distributions, where the
libggml library most likely will be shipped as an individual package
upon which a llama.cpp package depends.

ggerganov commented on cmake/common.cmake:
```diff
@@ -1,3 +1,5 @@
+include("ggml/cmake/common.cmake")
```
I think it would be better to have `llama.cpp`-specific flags instead of reusing the `ggml` flags like this. But this also works.

ckastner replied:
I agree. I didn't know whether the re-use of `ggml_get_flags` was intentional or not, so I went with the least intrusive option. Well, it can easily be changed once the need arises.
